test/estimate-gas.test.js

```diff
@@ -0,0 +1,138 @@
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'register device', gas: gas } )
+        })
+        it('   ^= to Register and Activate Device', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, (regFee + actFee), { from: ctx.actors.mfg })
+            var gas = await ctx.contracts.atonomi.registerAndActivateDevice.estimateGas(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'reg & act device', gas: gas } )
+        })
+        it('   ^= to Withdraw Tokens', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee + actFee, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.registerAndActivateDevice(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.updateReputationScore(deviceId, score, {from: ctx.actors.irnNode})
+            const testWithdraw = { account: ctx.actors.mfg, expectedTokens: mfgReward }
+            const from = testWithdraw.account
+            var gas  = await ctx.contracts.atonomi.withdrawTokens.estimateGas({ from: from })
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'withdraw tokens', gas: gas } )
+        })
+    })
+
+    describe('owner costs', () => {
+        it('   ^= to Activate Device', async() => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee, { from: ctx.actors.mfg })
+            await ctx.contracts.atonomi.registerDevice(deviceIdHash, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, actFee, { from: ctx.actors.deviceOwner })
+            var gas = await ctx.contracts.atonomi.activateDevice.estimateGas(deviceId, {from: ctx.actors.deviceOwner})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'owner', task: 'activate device', gas: gas } )
+        })
+    })
+    
+    describe('reputation author costs', () => {
+        it('   ^= to Submit Reputation', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee + actFee, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.registerAndActivateDevice(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            const gas = await ctx.contracts.atonomi.updateReputationScore.estimateGas(deviceId, score, {from: ctx.actors.irnNode})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'reputation author', task: 'submit reputation', gas: gas } )
+        })
+        it('   ^= to Withdraw Tokens', async() => {
```

Comment:
This is duplicate?  I think we can remove this one.

Reply:
good call - i changed it to calculate once and store and use it again